lib/developer_portal/app/views/developer_portal/javascripts/3scale.js

```diff
@@ -701,36 +701,31 @@ function visibility_condition_for(target_name, decider_name, value) {
        synchronize_checkboxes();
      })();
 
-    (function(){
-      if ($('#plan-select').length == 0) return;
-	var currentPlanID = $('#plans-selector').attr('data-plan-id');
-	var $plans = $('div.plan-preview');
-
-	$('#plan-select')[0].options[0].value
-	var options = $('#plan-select')[0].options;
-
-	for (var i = options.length - 1; i >= 0; i--){
-	  if(options[i].value == currentPlanID) {
-	    options.selectedIndex = (i - length);
-	  }
-	};
-
-      function attachEvents(){
-	$('#plan-select').change(function(){
-
-	  var planID = this.options[this.selectedIndex].value;
-	    $plans.hide();
-	    $('div.plan-preview[data-plan-id="'+planID+'"]').show();
-
-	    // HACK HACK HACK - redo the plan selector!
-	    if ($('#plans-selector').attr('data-plan-id') == planID) {
-	      $('#plan-change-submit').hide();
-	    } else {
-	      $('#plan-change-submit').show();
-	    }
-
-	    return false;
-	});
+    (function() {
```

Comment:
This JavaScript was broken. It did not actually select the current plan from the dropdown (`$('#plans-selector').attr('data-plan-id');` is actually undefined), so the first element from the list was always selected.

This caused an issue when the default scope "ordered by position" was removed. I guess what happened was that the plan that was supposed to be selected in the test appeared on the first position, and so as it was already selected the "changed" event never fired, and the "Change plan" button was never made visible.

Locally I could only reproduce on PostgreSQL 10, PostgreSQL 13 was fine - well, the order in the absence of `ORDER BY` is unpredictable, so OK.

As this code belongs to the developer portal, all the existing portals will still be broken. But luckily, I think this is a super rarely used feature (change or request changing the **account** plan), so I wouldn't worry about it. The main purpose of this refactor is to make tests work fine :)

Reply:
I think it was broken 13 years ago when someone changed `id=plan-select` with `class=plan-select` 🤦🏻 https://github.com/3scale/system/commit/5eb3ff511b84a8b706dc0c3728b99f64fcd143f1

Reply:
@josemigallas I think actually it was broken by this one: https://github.com/3scale/system/commit/87c94ccc6809e7dc334bf4d3ef547974368cba46#diff-e5a5ec089dcf3da6befa22832c910e836095d0bfacbfe24f4e834d354b4f59c1L3

And now looking at it, bringing it back probably would be a better fix - cause `$('.plan-preview:not([style*="display:none"])').first().attr('data-plan-id');` is not the nicest query in the world... 

But well, this behavior works well, and I don't think we need to waste more time on this :grimacing: